src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -99,9 +99,11 @@ public static object GetUninitializedObjectWithFormatterServices(Type type)
         private static List<IExternalSerializer> externalSerializers;
         private static ConcurrentDictionary<Type, IExternalSerializer> typeToExternalSerializerDictionary;
         private static Dictionary<string, Type> types;
-        private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
-        private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
-        private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
+        volatile private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
+        volatile private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
+        volatile private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
```

Comment:
if you are worried about cases that reads need to be up to date entirely and cannot use a stale cache (which will not stay forever), then use Volatile.Read(ref copiers) instead. In my experience, using volatile was almost always the wrong (and lazy) approach. I remember even seeing lots of smart people discussing that introducing the `volatile` keyword in c# was a bad idea because it does not do the intuitive thing. I don't recall the details, but let me know if you'd like me to try to find it. Regardless, Volatile.Read might ease your concerns.